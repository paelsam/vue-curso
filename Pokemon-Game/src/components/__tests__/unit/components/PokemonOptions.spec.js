import { shallowMount } from '@vue/test-utils'
import { describe, test, beforeEach, expect } from 'vitest'
import PokemonOptions from '@/components/PokemonOptions.vue'
import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonOptions Component', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallowMount( PokemonOptions, {
            props: {
                pokemons,
           }
        })
    })

    test('debe de hacer match con el snapshot', () => { 
        console.log( wrapper.html() );
        // toMatchInlineSnapshot
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('debe de mostrar las 4 opciones correctamente', () => {
        const $li = wrapper.findAll('li')
        expect( $li.length ).toBe( 4 )
        
        expect( $li[0].text() ).toBe('Bulbasaur')
        expect( $li[1].text() ).toBe('Ivysaur')
        expect( $li[2].text() ).toBe('Venusaur')
        expect( $li[3].text() ).toBe('Charmander')
    })

    test('debe de emitir "selection" con sus respectivos parÃ¡metros al hacer click', () => {

        const [ l1, l2, l3, l4 ] = wrapper.findAll('li')
        
        l1.trigger('click')
        l2.trigger('click')
        l3.trigger('click')
        l4.trigger('click')

        // Valores emitidos
        console.log( wrapper.emitted('selection') )

        expect( wrapper.emitted('selection').length ).toBe(4)
        expect( wrapper.emitted('selection')[0] ).toStrictEqual([1])
        expect( wrapper.emitted('selection')[1] ).toStrictEqual([2])
        expect( wrapper.emitted('selection')[2] ).toStrictEqual([3])
        expect( wrapper.emitted('selection')[3] ).toStrictEqual([4])




    })
})