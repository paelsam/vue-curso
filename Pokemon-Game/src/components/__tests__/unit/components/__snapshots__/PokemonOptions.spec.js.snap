// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonOptions Component > debe de hacer match con el snapshot 1`] = `
"<div data-v-39189cb9=\\"\\" class=\\"options-container\\">
  <ul data-v-39189cb9=\\"\\">
    <li data-v-39189cb9=\\"\\">Bulbasaur</li>
    <li data-v-39189cb9=\\"\\">Ivysaur</li>
    <li data-v-39189cb9=\\"\\">Venusaur</li>
    <li data-v-39189cb9=\\"\\">Charmander</li>
  </ul>
</div>"
`;
